zkevm-circuits/src/evm_circuit/op_execution/utils/math_gadgets.rs

```diff
@@ -83,3 +85,146 @@ impl<F: FieldExt> IsEqualGadget<F> {
         self.is_zero.assign(region, offset, lhs - rhs)
     }
 }
+
+/// Returns `1` when `lhs < rhs`, and returns `0` otherwise.
+/// lhs and rhs `< 256**NUM_BYTES`
+/// `NUM_BYTES` is required to be `<= MAX_BYTES_FIELD` to prevent overflow:
+/// values are stored in a single field element and two of these
+/// are added together.
+/// The equation that is enforced is `lhs == rhs + diff - (lt * range)`.
+/// Because all values are `<= 256**NUM_BYTES` and `lt` is boolean,
+///  `lt` can only be `1` when `lhs < rhs`.
+#[derive(Clone, Debug)]
+pub struct LtGadget<F, const NUM_BYTES: usize> {
+    lt: Cell<F>, // `1` when `lhs < rhs`, `0` otherwise.
+    diff: [Cell<F>; NUM_BYTES], /* The byte values of `diff`.
+                  * `diff` equals `lhs - rhs` if `lhs >= rhs`,
+                  * `lhs - rhs + range` otherwise. */
+    range: F, // The range of the inputs, `256**NUM_BYTES`
+}
+
+impl<F: FieldExt, const NUM_BYTES: usize> LtGadget<F, NUM_BYTES> {
+    pub const NUM_CELLS: usize = 1 + NUM_BYTES; // lt + diff
+    pub const NUM_WORDS: usize = 0;
+
+    pub(crate) fn construct(alloc: &mut CaseAllocation<F>) -> Self {
+        assert!(NUM_BYTES <= MAX_BYTES_FIELD, "unsupported number of bytes");
+        Self {
+            lt: alloc.cells.pop().unwrap(),
+            diff: array_init(|_| alloc.cells.pop().unwrap()),
+            range: utils::get_range(NUM_BYTES * 8),
+        }
+    }
+
+    pub(crate) fn constraints(
+        &self,
+        cb: &mut ConstraintBuilder<F>,
+        lhs: Expression<F>,
+        rhs: Expression<F>,
+    ) -> Expression<F> {
+        let diff = utils::from_bytes::expr(self.diff.to_vec());
+        // The equation we require to hold: `lhs == rhs + diff - (lt * range)`.
+        cb.require_equal(lhs, rhs + diff - (self.lt.expr() * self.range));
```

Comment:
Though they are the same, I feel it looks more clear changing it this way.

```suggestion
        cb.require_equal(lhs - rhs, diff - (self.lt.expr() * self.range));
```

Reply:
Agree!